common/src/main/java/com/wynntils/functions/generic/ConditionalFunctions.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.functions.generic;
+
+import com.wynntils.core.functions.GenericFunction;
+import com.wynntils.core.functions.arguments.FunctionArguments;
+import java.util.List;
+
+public class ConditionalFunctions {
+    public static class IfStringFunction extends GenericFunction<Object> {
```

Comment:
Hm. This looks like it could do with a bit of generics. Create an abstract IfFunction<T> and just let IfStringFunction extend IfFunction<String>. Relace the String.class with T in the argument builder. And then you can also replace the return value of getValue to T.

Reply:
Or, as things look rn, I don't even see why you have two version. Both have getValue that returns an object. Maybe you should instead allow for a general argument parser that can take a string as well as a number, and just returns an object?

Reply:
> Or, as things look rn, I don't even see why you have two version. Both have getValue that returns an object. Maybe you should instead allow for a general argument parser that can take a string as well as a number, and just returns an object?

I am quite happy with how it is right now, except for this singular function... I don't think I should design the whole parser around one of the most "edge case" functions. So, I can generify this, will you accept that?